src/node/transaction.cpp

```diff
@@ -101,7 +101,7 @@ TransactionError BroadcastTransaction(NodeContext& node, const CTransactionRef t
                 // with a transaction to/from their wallet, immediately call some
                 // wallet RPC, and get a stale result because callbacks have not
                 // yet been processed.
-                CallFunctionInValidationInterfaceQueue([&promise] {
+                node.validation_signals->CallFunctionInValidationInterfaceQueue([&promise] {
                     promise.set_value();
                 });
                 callback_set = true;
```

Comment:
In 38d32ba2:

What about setting `callback_set=false` if `validation_signals` is null?

Reply:
I'm not sure if this should be optional. Was thinking before about asserting here. Wdyt?

Reply:
> I'm not sure if this should be optional. Was thinking before about asserting here. Wdyt?

I don't see a benefit to aborting when the wait_callback option is true and there is no callback. There is no callback if the transaction is already in the mempool, and we are ok with ignoring the wait_callback option in that case. Would suggest changing line 95 to `if (wait_callback && node.validation_signals)`